app/metrics.yaml

```
@@ -1856,3 +1856,21 @@ installation:
     notification_emails:
       - fenix-core@mozilla.com
     expires: "2020-09-01"
+
+startup.timeline.telemetry:
```

```suggestion
startup.perf.timeline:
```

I see your point about naming confusion. The metrics are generated within the `org.mozilla.fenix.GleanMetrics` package, so they would end up in `org.mozilla.fenix.GleanMetrics.StartupTimeline`. In the future, having `telemetry` as a suffix will produce a build time error :-)

---

How about `startup.timeline.metrics` or `startup.timeline.measurements`? I think the existing `StartupTimeline` could easily be confused with `StartupPerfTimeline`, e.g. when looking through an autocomplete menu.

fwiw, in Java, I don't think it's uncommon for the package name to have redundancies with the class name because the class is often referenced without the package. For example, `androidx.fragment.app.Fragment`.

---

`metrics` and `measurements` would still be problematic, [see our docs](https://mozilla.github.io/glean/book/user/adding-new-metrics.html#be-as-specific-as-possible).

Telemetry, metrics and measurements are all a bit redundant :-)

---

I see: it's redundant in the end-to-end system:

> It is not necessary to put the type of the metric in the category or name, since this information is retained in other ways through the entire end-to-end system.

But from a developer perspective it's not redundant when reading only the Java class name. One way to resolve this problem is that Glean's generated methods could add one of these endings, e.g. in YAML: `startup.timeline` = in Java `StartupTimelineMetrics`.

That being said, this is what we have now so maybe I'll stick with `StartupTimeline` (as the most accurate value and shortest for upload through the pipeline) and educate the small number of developers using the code that there's a namespace collision.

---

Made the changed but keeping unresolved so we can continue the conversation.

---

> > It is not necessary to put the type of the metric in the category or name, since this information is retained in other ways through the entire end-to-end system.
> 
> But from a developer perspective it's not redundant when reading only the Java class name. One way to resolve this problem is that Glean's generated methods could add one of these endings, e.g. in YAML: `startup.timeline` = in Java `StartupTimelineMetrics`.

That's true. If we were to allow this I'm afraid we'd need to consider all the other platforms/bindings the Glean SDK is currently available on. Some bindings might not support aliasing at all, I'm afraid :-( On a quick look, it seems that complexity outweights advantages on this one (but I might be wrong).

Please kindly [file a bug](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data+Platform+and+Tools&component=Glean%3A+SDK) about this if you feel strongly about it :-) 

> That being said, this is what we have now so maybe I'll stick with `StartupTimeline` (as the most accurate value and shortest for upload through the pipeline) and educate the small number of developers using the code that there's a namespace collision.

Another option that we did not consider was to rename `StartupTimeline.kt` to somethine like `StartupTimelineMonitor.kt` (or similar). Would this make it clearer?

---

I filed a bug https://bugzilla.mozilla.org/show_bug.cgi?id=1624066 though I'm feeling less sure about it because the name of the metric gets longer, making it more tedious to use. We should get more feedback.

> Another option that we did not consider was to rename StartupTimeline.kt to somethine like StartupTimelineMonitor.kt (or similar). Would this make it clearer?

We _could_ do this to avoid the collision but:
- `StartupTimeline` is a name from desktop so I wanted to keep them consistent
- `StartupTimelineMonitor` gets long and annoying to call statically (i.e. the same problem I realized above)
- If someone sees `StartupTimelineMonitor`, they might be confused what the class `StartupTimeline` does – does the monitor monitor this other class in some way?

That being said, thinking about it now, `StartupTimeline` is a confusing name on it's own and it should really be `StartupTimelineMonitor` 🤔. This is confusing for a Friday evening: I'll just leave everything where it is now. :)

---

> > Another option that we did not consider was to rename StartupTimeline.kt to somethine like StartupTimelineMonitor.kt (or similar). Would this make it clearer?
>
> We _could_ do this to avoid the collision but:
> 
>     * `StartupTimeline` is a name from desktop so I wanted to keep them consistent

I would not stick to that too much. That might change as well, once glean comes to desktop.

>     * `StartupTimelineMonitor` gets long and annoying to call statically (i.e. the same problem I realized above)

That could be a problem, but do you expect to use this more than 10/15 times throughout the code?

>     * If someone sees `StartupTimelineMonitor`, they might be confused what the class `StartupTimeline` does – does the monitor monitor this other class in some way?

This could be eased by comments, no?

---

> I would not stick to that too much. That might change as well, once glean comes to desktop.

That's fair.

> That could be a problem, but do you expect to use this more than 10/15 times throughout the code?

10/15 seems like the minimum: e.g. we'd likely instrument each of the major lifecycle methods in each major activity and fragment at the start and end of the method (that's ~16 already).

> This could be eased by comments, no?

True but it's better to avoid this if possible. :)

---

> > That could be a problem, but do you expect to use this more than 10/15 times throughout the code?
> 
> 10/15 seems like the minimum: e.g. we'd likely instrument each of the major lifecycle methods in each major activity and fragment at the start and end of the method (that's ~16 already).

This could further be eased by import aliasing, e.g. `import org.mozilla.fenix.perf.StartupTimelineMonitor as stm` or something like that if we come to that, right?